src/middlewared/middlewared/plugins/network.py

```diff
@@ -584,23 +608,45 @@ def iface_extend(self, iface_state, configs, fake=False):
 
         await self.__save_datastores()
 
-        if data['type'] == 'LINK_AGGREGATION':
-            name = data.get('name')
-            if not name:
-                next_lagg = 0
-                laggs = [
-                    i['int_interface']
-                    for i in await self.middleware.call(
-                        'datastore.query',
-                        'network.interfaces',
-                        [('int_interface', '^', 'lagg')],
-                    )
-                ]
-                while f'lagg{next_lagg}' in laggs:
-                    next_lagg += 1
-                name = f'lagg{next_lagg}'
+        async def get_next(prefix, start=0):
+            number = start
+            ifaces = [
+                i['int_interface']
+                for i in await self.middleware.call(
+                    'datastore.query',
+                    'network.interfaces',
+                    [('int_interface', '^', prefix)],
+                )
+            ]
+            while f'{prefix}{number}' in ifaces:
+                number += 1
+            return f'{prefix}{number}'
 
-            interface_id = None
+        interface_id = None
+        if data['type'] == 'BRIDGE':
+            # For bridge we want to start with 2 because bridge0/bridge1 may have been used
+            # for Jails/VM.
+            name = data.get('name') or await get_next('bridge', start=2)
+            try:
+                async for i in self.__create_interface_datastore(data, {
+                    'interface': name,
+                }):
+                    interface_id = i
+
+                await self.middleware.call(
+                    'datastore.insert',
+                    'network.bridge',
+                    {'interface': interface_id, 'members': data['bridge_members']},
+                )
+            except Exception as e:
+                if interface_id:
+                    with contextlib.suppress(Exception):
+                        await self.middleware.call(
+                            'datastore.delete', 'network.interfaces', interface_id
+                        )
+                raise e
```

Comment:
Just `raise`

Reply:
No it won't
```python
import contextlib
try:
    1 / 0
except Exception:
    with contextlib.suppress(Exception):
        2 / 0
    raise
```

```
[desktop] ~> python3 a.py
Traceback (most recent call last):
  File "a.py", line 3, in <module>
    1 / 0
ZeroDivisionError: division by zero
```

Reply:
Right, my bad. But what difference does it make? ;)